src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -113,28 +96,31 @@ public void storeMessage(Message message) {
     // All the messages between the two users
     Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
 
+    // Sort in the ascending order so that most recent messages appear last
```

Comment:
Thanks for adding the comment. One additional piece of feedback related to this:

In terms of making comments valuable, focus on _why_ something is being done, not _what is being done_. That "most recent messages appear last" describes what is happening, but not why you want it to happen.

I think it would make be more helpful to do:

      Query query =
         new Query("Message")
            .setFilter(directMessages)
            // Messages are ordered from oldest to newest since messages between
            // people may include implicit context/reference to previously sent messages,
            // and this order makes the chronology of the conversation and context
            // more intuitive and understandable in the messages list page. 
            .addSort("timestamp", SortDirection.ASCENDING);